ui/src/app/components/common-env-conf-panels/start-stop-environment-button.tsx

```diff
@@ -20,143 +19,101 @@ import computeStarting from 'assets/icons/compute-starting.svg';
-    [
-      UserEnvironmentStatus.CREATING,
-      () => ({
-        altText: 'Environment creation in progress',
-        iconSrc: computeStarting,
-        dataTestId: 'environment-status-icon-starting',
-        styleOverrides: rotateStyle,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.RUNNING,
-      () => ({
-        altText: 'Environment running, click to pause',
-        iconSrc: computeRunning,
-        dataTestId: 'environment-status-icon-running',
-        onClick: onPause,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.UPDATING,
-      () => ({
-        altText: 'Environment update in progress',
-        iconSrc: computeStarting,
-        dataTestId: 'environment-status-icon-starting',
-        styleOverrides: rotateStyle,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.ERROR,
-      () => ({
-        altText: 'Environment in error state',
-        iconSrc: computeError,
-        dataTestId: 'environment-status-icon-error',
-      }),
-    ],
-    [
-      UserEnvironmentStatus.PAUSING,
-      () => ({
-        altText: 'Environment pause in progress',
-        iconSrc: computeStopping,
-        dataTestId: 'environment-status-icon-stopping',
-        styleOverrides: rotateStyle,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.PAUSED,
-      () => ({
-        altText: 'Environment paused, click to resume',
-        iconSrc: computeStopped,
-        dataTestId: 'environment-status-icon-stopped',
-        onClick: onResume,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.RESUMING,
-      () => ({
-        altText: 'Environment resume in progress',
-        iconSrc: computeStarting,
-        dataTestId: 'environment-status-icon-starting',
-        styleOverrides: rotateStyle,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.DELETING,
-      () => ({
-        altText: 'Environment deletion in progress',
-        iconSrc: computeStopping,
-        dataTestId: 'environment-status-icon-stopping',
-        styleOverrides: rotateStyle,
-      }),
-    ],
-    [
-      UserEnvironmentStatus.DELETED,
-      () => ({
-        altText: 'Environment has been deleted',
-        iconSrc: computeNone,
-        dataTestId: 'environment-status-icon-none',
-      }),
-    ],
-    [
-      UserEnvironmentStatus.UNKNOWN,
-      () => ({
-        altText: 'Environment status unknown',
-        iconSrc: computeNone,
-        dataTestId: 'environment-status-icon-none',
-      }),
-    ],
-    [
-      DEFAULT,
-      () => ({
-        altText: 'No Environment found',
-        iconSrc: computeNone,
-        dataTestId: 'environment-status-icon-none',
-      }),
-    ]
-  );
 
-  {
-    /* height/width of the icon wrapper are set so that the img element can rotate inside it */
-  }
-  {
-    /* without making it larger. the svg is 36 x 36 px, per pythagorean theorem the diagonal */
-  }
-  {
-    /* is 50.9px, so we round up */
-  }
+  const toProps: Record<UserEnvironmentStatus, ButtonProps> = {
```

Comment:
switchCase -> Record

Reply:
Yeah I learned about it recently ... I think there are a few other places we can put this to good use.

I think the brackets are needed because this is an Object, and the rules of what can be an Object key requires this, if it's not a plain string